--- jvmcommon.sh.orig   2021-07-28 15:20:16.268279233 +0000
+++ jvmcommon.sh        2021-07-28 15:31:51.522736687 +0000
@@ -3,20 +3,22 @@
 calculate_java_memory_opts() {
   local opts=${1:-""}
 
-  limit=$(ulimit -u)
+  limit=$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)
   case $limit in
-  512) # 2X, private-s: memory.limit_in_bytes=1073741824
+  1073741824) # 2X, private-s: memory.limit_in_bytes=1073741824
     echo "$opts -Xmx671m -XX:CICompilerCount=2"
     ;;
-  16384) # perf-m, private-m: memory.limit_in_bytes=2684354560
+  2684354560) # perf-m, private-m: memory.limit_in_bytes=2684354560
     echo "$opts -Xmx2g"
     ;;
-  32768) # perf-l, private-l: memory.limit_in_bytes=15032385536
+  15032385536) # perf-l, private-l: memory.limit_in_bytes=15032385536
     echo "$opts -Xmx12g"
     ;;
-  *) # Free, Hobby, 1X: memory.limit_in_bytes=536870912
+  536870912) # Free, Hobby, 1X: memory.limit_in_bytes=536870912
     echo "$opts -Xmx300m -Xss512k -XX:CICompilerCount=2"
     ;;
+  *) # Some other environment
+    echo "$opts"
   esac
 }
 
@@ -35,9 +37,9 @@
 fi
 
 if [ -f "$JAVA_HOME/release" ] && grep -q '^JAVA_VERSION="1[0-9]' "$JAVA_HOME/release"; then
-  default_java_mem_opts="$(calculate_java_memory_opts "-XX:+UseContainerSupport")"
+  default_java_mem_opts="$(calculate_java_memory_opts "-XX:+UseContainerSupport -XX:+UseShenandoahGC -XX:ShenandoahGCHeuristics=compact -XX:+UseStringDeduplication -XX:MaxRAMFraction=1 -XX:+ExitOnOutOfMemoryError")"
 else
-  default_java_mem_opts="$(calculate_java_memory_opts | sed 's/^ //')"
+  default_java_mem_opts="$(calculate_java_memory_opts "-XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport -XX:+UseG1GC -XX:+UseStringDeduplication" | sed 's/^ //')"
 fi
 
 if echo "${JAVA_OPTS:-}" | grep -q "\-Xmx"; then