--- jvmcommon.sh.old	2025-02-10 18:41:48.128537300 +0100
+++ jvmcommon.sh	2025-02-10 18:43:18.960393100 +0100
@@ -59,7 +59,7 @@
 fi
 
 if [ -f "$JAVA_HOME/release" ] && grep -q '^JAVA_VERSION="1[0-9]' "$JAVA_HOME/release"; then
-	default_java_mem_opts="$(calculate_java_memory_opts "-XX:+UseContainerSupport")"
+	default_java_mem_opts="$(calculate_java_memory_opts "-XX:+UseContainerSupport -XX:+UseShenandoahGC -XX:ShenandoahGCHeuristics=compact -XX:+UseStringDeduplication -XX:+ExitOnOutOfMemoryError")"
 else
 	default_java_mem_opts="$(calculate_java_memory_opts | sed 's/^ //')"
 fi
