--- jvmcommon.sh.orig	2025-03-20 13:29:37.700911600 +0100
+++ jvmcommon.sh	2025-03-20 13:39:43.459615200 +0100
@@ -47,7 +47,7 @@
 	1073741824) flags+=("-Xmx671m" "-XX:CICompilerCount=2") ;;
 	# Rely on JVM ergonomics for other dyno types, but increase the maximum RAM percentage from 25% to 80%.
 	# This is consistent with the historic Heroku defaults for dyno types not listed above.
-	*) flags+=("-XX:MaxRAMPercentage=80.0") ;;
+	*) ;;
 	esac
 
 	(
@@ -56,7 +56,7 @@
 	)
 }
 
-jvm_options="$(jvm_options)"
+jvm_options="$(jvm_options) -XX:+UseContainerSupport -XX:+UseShenandoahGC -XX:ShenandoahGCHeuristics=compact -XX:+UseStringDeduplication -XX:+ExitOnOutOfMemoryError"
 export JAVA_OPTS="${jvm_options}${JAVA_OPTS:+" "}${JAVA_OPTS:-}"
 
 if ! [[ "${DYNO}" =~ ^run\..*$ ]]; then
